be/src/vec/functions/function_width_bucket.cpp

```diff
@@ -0,0 +1,156 @@
+        ColumnPtr min_value_ptr =
+                block.get_by_position(arguments[1]).column->convert_to_full_column_if_const();
+        ColumnPtr max_value_ptr =
+                block.get_by_position(arguments[2]).column->convert_to_full_column_if_const();
+        ColumnPtr num_buckets_ptr =
+                block.get_by_position(arguments[3]).column->convert_to_full_column_if_const();
+        int64_t num_buckets = num_buckets_ptr->get_int(0);
+
+        auto nested_column_ptr = ColumnInt64::create(input_rows_count, 0);
+        DataTypePtr expr_type = block.get_by_position(arguments[0]).type;
+
+        _execute_by_type(*expr_ptr, *min_value_ptr, *max_value_ptr, num_buckets, *nested_column_ptr,
+                         expr_type);
+
+        WhichDataType which(remove_nullable(expr_type));
+        if (which.is_nullable()) {
+            auto dest_column_ptr =
+                    ColumnNullable::create(std::move(nested_column_ptr),
+                                           ColumnUInt8::create(nested_column_ptr->size(), 0));
+            block.replace_by_position(result, std::move(dest_column_ptr));
+        } else {
+            block.replace_by_position(result, std::move(nested_column_ptr));
+        }
+        return Status::OK();
+    }
+
+private:
+    template <typename ColumnType>
+    void _execute(const IColumn& expr_column, const IColumn& min_value_column,
+                  const IColumn& max_value_column, const int64_t num_buckets,
+                  IColumn& nested_column) {
+        const ColumnType& expr_column_concrete = reinterpret_cast<const ColumnType&>(expr_column);
+        const ColumnType& min_value_column_concrete =
+                reinterpret_cast<const ColumnType&>(min_value_column);
+        const ColumnType& max_value_column_concrete =
+                reinterpret_cast<const ColumnType&>(max_value_column);
+        ColumnInt64& nested_column_concrete = reinterpret_cast<ColumnInt64&>(nested_column);
+
+        auto min_value = min_value_column_concrete.get_data()[0];
+        auto max_value = max_value_column_concrete.get_data()[0];
+
+        size_t input_rows_count = expr_column.size();
+
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            if (expr_column_concrete.get_data()[i] < min_value) {
+                continue;
+            } else if (expr_column_concrete.get_data()[i] >= max_value) {
+                nested_column_concrete.get_data()[i] = num_buckets + 1;
+            } else {
+                if ((max_value - min_value) / num_buckets == 0) {
+                    continue;
+                }
+                nested_column_concrete.get_data()[i] =
+                        (int64_t)(1 + (expr_column_concrete.get_data()[i] - min_value) /
+                                              ((max_value - min_value) / num_buckets));
+            }
+            LOG(WARNING) << std::to_string(i) + "   " +
```

Comment:
Do not print WARNING log here

Reply:
This is used for debugging. I forgot to remove it.